vcell-core/src/main/java/cbit/vcell/mapping/SpeciesContextSpec.java

```diff
@@ -70,8 +70,14 @@ public class SpeciesContextSpec implements Matchable, ScopedSymbolTable, Seriali
     public static final String PARAMETER_NAME_PROXY_PARAMETERS = "proxyParameters";
     private static final String PROPERTY_NAME_WELL_MIXED = "wellMixed";
     private static final String PROPERTY_NAME_FORCECONTINUOUS = "forceContinuous";
-    public static final String PROPERTY_NAME_SITE_ATTRIBUTE = "SiteAttributes";     // springsalad
-    public static final String PROPERTY_NAME_LINK_LENGTH = "LinkLength";            // springsalad
+    // springsalad group
+    public static final String PROPERTY_NAME_SITE_ATTRIBUTE = "SiteAttributes";
+    public static final String PROPERTY_NAME_LINK_LENGTH = "LinkLength";
+    public static final String PROPERTY_NAME_SITE_SELECTED_IN_TABLE = "SiteSelectedInTable";
+    public static final String PROPERTY_NAME_LINK_SELECTED_IN_TABLE = "LinkSelectedInTable";
+    public static final String PROPERTY_NAME_SITE_SELECTED_IN_SHAPE = "SiteSelectedInShape";

```

Comment:
for these 4 new bound properties, make sure you un-listen and re-listen where appropriate to (1) keep the UI responsive even after add/remove of objects and (2) prevent memory leaks and invisible components trying to update themselves.